scripts/docgen/snippets/extractSnippets.ts

```diff
@@ -103,14 +104,11 @@ const extractLabeledSnippets = (sourceFileText: string): LabeledSnippets => {
                     matchingSnipStartIndex,
                     1
                 )[0]
-                text = sourceFileText.slice(matchingSnipStart.lineNumber, i + 1)
-                text = extractTextBetweenLines(
-                    textSplitByNewLine,
-                    matchingSnipStart.lineNumber,
-                    i
+                text = linesToOutput(
+                    lines.slice(matchingSnipStart.lineNumber, lineNumber)
                 )
             } else if ((parsedSnip.kind = "@snipLine")) {
-                text = extractTextBetweenLines(textSplitByNewLine, i, i + 1)
+                text = linesToOutput(lines.slice(lineNumber, lineNumber + 1))
             }
             labeledSnippets[parsedSnip.id] = { text: text as string }
```

Comment:
Sorry I missed this the first time, but similarly to the previous comment you addressed about casting `parsedSnip`, TS should be able to infer `text` is a string here. If it doesn't, it probably means that somehow it could actually be undefined in which case we'd want to make sure we figure out when and address it.